app/models/job.rb

```diff
@@ -36,6 +36,19 @@ class Job
   attr_accessor :script
   attr_accessor :state
 
+  # Handle the k and m suffix
+  attr_reader :min_nodes
+  def min_nodes=(num)
+    str = num.to_s
+    @min_nodes ||= if /\A\d+k\Z/.match?(str)
+      str.sub('k', '').to_i * 1024
+    elsif /\A\d+m\Z/.match(str)
+      str.sub('m', '').to_i * 1048576
+    else
+      num
```

Comment:
If we change this to `num.to_i` I think we can remove the `to_i` from the various `node.min_nodes.to_i` calls elsewhere.  That way once the value has been set on the Job we're always dealing with a number afterwards.

Reply:
I was contemplating that, but I don't know how nicely it will handle `min_num: "some-bogus-string"`. It will essentially defeat the purpose of the validation

Reply:
OK. I see what you mean.  Let's leave it as it for now.  I may have a think on this some more.

Reply:
Just realised we can use `regex` (like the other regex) :facepalm: 

https://github.com/openflighthpc/flight-scheduler-controller/pull/2/commits/3c129c69432c782950ac1a2c3ef7d31c85e220b4